src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverter.cs

```diff
@@ -64,7 +56,9 @@ internal virtual object CreateObject(JsonSerializerOptions options)
             JsonSerializerOptions options,
             ref ReadStack state)
         {
-            throw new InvalidOperationException(SR.NodeJsonObjectCustomConverterNotAllowedOnExtensionProperty);
+            Debug.Fail("Should not be reachable.");
```

Comment:
can this method be `private protected`?

Reply:
We'd need to override it explicitly in every derived `JsonConverter` so probably not.